luarules/gadgets/unit_collision_damage_behavior.lua

```diff
@@ -0,0 +1,171 @@
+function gadget:GetInfo()
+	return {
+		name = "Collision Damage Behavior",
+		desc = "Magnifies the default engine ground and object collision damage and handles max impulse limits",
+		author = "SethDGamre",
+		date = "2024.8.29",
+		license = "GNU GPL, v2 or later",
+		layer = 0,
+		enabled = true
+	}
+end
+
+if not gadgetHandler:IsSyncedCode() then return end
+
+--customparams.fall_damage_multiplier = <number> a multiplier that's applied to defaultDamageMultiplier which affects the amount of damage taken from falling/collisions.
+
+--the multiplier by which default engine ground/object collision damage is multiplied. change this value to reduce the amount of fall/collision damage taken for all units.
+local fallDamageMagnificationFactor = 14
+
+--this defines how fast a unit has to be moving in order to take object collision damage
+local collisionVelocityThreshold = 3.3
+
+-- Decrease this value to make units move less from impulse. This defines the maximum impulse allowed, which is (maxImpulseMultiplier * mass) of each unit.
+local maxImpulseMultiplier = 5
+
+--measured in elmos per frame. If a unit is launched via explosion faster than this, it is instantly slowed to this value
+local velocityCap = 9
+
+--measured in elmos per frame. If velocity is above this threshold, it will be slowed until below this threshold.
+local velocitySlowdownThreshold = 6
+
+--any weapondef impulseFactor below this is ignored to save performance
+local minImpulseFactor = 0.15
+
+--any impulse to damage ratio below this is ignored to save performance
+local minImpulseToDamageRatio = 0.2
+
+local groundCollisionDefID = Game.envDamageTypes.GroundCollision
+local objectCollisionDefID = Game.envDamageTypes.ObjectCollision
+local spGetUnitHealth = Spring.GetUnitHealth
+local spGetUnitVelocity = Spring.GetUnitVelocity
+local spSetUnitVelocity = Spring.SetUnitVelocity
+local mathMin = math.min
+local mathMax = math.max
+
+local fallDamageMultipliers = {}
+local unitsMaxImpulse = {}
+local weaponDefIDImpulses = {}
+local transportedUnits = {}
+local unitMasses = {}
+local weaponDefIgnored = {}
+local unitInertiaCheckFlags = {}
+local gameFrame = 0
+local velocityWatchDuration = 8
```

Comment:
8s or 8f?

Reply:
Frames

Reply:
I know, what I'm saying is to make it so that the next guy reading this code won't have to ask.

Reply:
got it, renamed to velocityWatchFrames